openapi/pacta.yaml

```diff
@@ -838,6 +876,66 @@ components:
         asset_id:
           type: string
           description: A unique identifier for the uploaded asset
+    StartPortfolioUploadRequest:
+      type: object
+      required:
+        - items
+      properties:
+        items:
+          type: array
+          items:
+            $ref: '#/components/schemas/StartPortfolioUploadRequestItem'
+    StartPortfolioUploadRequestItem:
+      type: object
+      required:
+        - file_name
+      properties:
+        file_name:
+          type: string
+          description: The name of the file, including its extension.
+    StartPortfolioUploadResponse:
+      type: object
+      required:
+        - items
+      properties:
+        items:
+          type: array
+          items:
+            $ref: '#/components/schemas/StartPortfolioUploadResponseItem'
+    StartPortfolioUploadResponseItem:
+      type: object
+      required:
+        - file_name
+        - upload_url
+        - incomplete_upload_id
+      properties:
+        file_name:
+          type: string
+          description: The name of the file, including its extension, used as a round-trip id.
+        upload_url:
+          type: string
+          description: The signed URL where the file should be uploaded to, using PUT semantics.
+        incomplete_upload_id:
+          type: string
+          description: A unique identifier for the uploaded asset
+    CompletePortfolioUpload:
+      type: object
+      required:
+        - incomplete_upload_ids
+      properties:
+        incomplete_upload_ids:
+          type: array
+          items:
+            type: string
```

Comment:
I'd consider having this be an object, like you did for `StartPortfolioUploadRequest.items`, just because you can't really add fields after this point to each upload_id (like a timestamp, or hash, or whatever) in a backwards compatible way otherwise.

Reply:
Great call.